src/util-systemd.c

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: MIT-0 */
+
+/* Implement the systemd notify protocol without external dependencies.
+ * Supports both readiness notification on startup and on reloading,
+ * according to the protocol defined at:
+ * https://www.freedesktop.org/software/systemd/man/latest/sd_notify.html
+ * This protocol is guaranteed to be stable as per:
+ * https://systemd.io/PORTABILITY_AND_STABILITY/ */
+
```

Comment:
No copyright? I know its in the commit, but a little note in the file as to where it came from would be ideal as well, if only to help with understanding code style as well.